gufe/tests/test_models.py

```diff
@@ -41,23 +69,135 @@ def test_default_settings():
     my_settings.schema_json(indent=2)
 
 
-@pytest.mark.parametrize(
-    "value,good",
-    [
-        ("parsnips", False),  # shouldn't be allowed
-        ("hbonds", True),
-        ("hangles", True),
-        ("allbonds", True),  # allowed options
-        ("HBonds", True),  # check case insensitivity
-    ],
-)
-def test_invalid_constraint(value, good):
-    if good:
-        s = OpenMMSystemGeneratorFFSettings(constraints=value)
-        assert s
-    else:
-        with pytest.raises(ValueError):
-            _ = OpenMMSystemGeneratorFFSettings(constraints=value)
+class TestSettingsValidation:
+    @pytest.mark.parametrize(
+        "value,valid,expected",
+        [
+            ("parsnips", False, None),  # shouldn't be allowed
+            ("hbonds", True, "hbonds"),
+            ("hangles", True, "hangles"),
+            ("allbonds", True, "allbonds"),  # allowed options
+            ("HBonds", True, "HBonds"),  # check case insensitivity TODO: cast this to lower?
+            (None, True, None),
+        ],
+    )
+    def test_openmmff_constraints(self, value, valid, expected):
+        if valid:
+            s = OpenMMSystemGeneratorFFSettings(constraints=value)
+            assert s.constraints == expected
+        else:
+            with pytest.raises(ValueError):
+                _ = OpenMMSystemGeneratorFFSettings(constraints=value)
+
+    @pytest.mark.parametrize(
+        "value,valid,expected",
+        [
+            (1.0 * unit.nanometer, True, 1.0 * unit.nanometer),
+            (1.0, True, 1.0 * unit.nanometer),  # should cast float to nanometer
```

Comment:
Another example of behavior we should change in 2.0, I don't think we should assume input units